// convert-units/lib/umd/definitions/speed.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var metric = {
        'm/s': {
            name: {
                singular: 'Metre per second',
                plural: 'Metres per second',
            },
            to_anchor: 3.6,
        },
        'km/h': {
            name: {
                singular: 'Kilometre per hour',
                plural: 'Kilometres per hour',
            },
            to_anchor: 1,
        },
    };
    var imperial = {
        mph: {
            name: {
                singular: 'Mile per hour',
                plural: 'Miles per hour',
            },
            to_anchor: 1,
        },
        knot: {
            name: {
                singular: 'Knot',
                plural: 'Knots',
            },
            to_anchor: 1.150779,
        },
        'ft/s': {
            name: {
                singular: 'Foot per second',
                plural: 'Feet per second',
            },
            to_anchor: 0.681818,
        },
    };
    var measure = {
        systems: {
            metric: metric,
            imperial: imperial,
        },
        anchors: {
            metric: {
                imperial: {
                    ratio: 1 / 1.609344,
                },
            },
            imperial: {
                metric: {
                    ratio: 1.609344,
                },
            },
        },
    };
    exports.default = measure;
});
