// @bundle-stats/utils/lib-esm/utils/date.js
import isDate from 'lodash/isDate';
const getNavigatorLanguages = () => (typeof window !== 'undefined' && typeof window.navigator !== 'undefined' && window.navigator.languages) || 'en';
const getResultAsString = (value) => {
    if (isDate(value)) {
        return value.toString();
    }
    if (typeof value === 'number') {
        return value.toString();
    }
    return value || '';
};
export const formatDateTime = (formatterOptions, value) => {
    let date;
    let result = '';
    if (!value) {
        return result;
    }
    if (isDate(value)) {
        date = value;
    }
    else {
        try {
            date = new Date(value);
        }
        catch (err) {
            return getResultAsString(value);
        }
    }
    const formatter = new Intl.DateTimeFormat(getNavigatorLanguages(), formatterOptions);
    try {
        result = formatter.format(date);
    }
    catch (err) {
        return getResultAsString(value);
    }
    return result;
};
export const formatDate = (value) => formatDateTime({
    day: '2-digit',
    month: '2-digit',
    year: 'numeric',
}, value);
export const formatTime = (value) => formatDateTime({
    ...(process.env.NODE_ENV !== 'test' && {
        hour: '2-digit',
    }),
    minute: '2-digit',
    second: '2-digit',
}, value);
