// @bundle-stats/utils/lib/webpack/extract/assets-cache-invalidation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.extractAssetsCacheInvalidation = void 0;

var _merge2 = _interopRequireDefault(require("lodash/merge"));

var _get2 = _interopRequireDefault(require("lodash/get"));

var _getMetricChanged = require("../../report/get-metric-changed");

var _mergeMetricsByKey = require("../../report/merge-metrics-by-key");

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const extractAssetsCacheInvalidation = (webpackStats, currentExtractedData, baselineBundleStats) => {
  const currentAssets = (0, _get2.default)(currentExtractedData, 'metrics.assets', {});
  const baselineAssets = (0, _get2.default)(baselineBundleStats, 'metrics.webpack.assets', {});
  const rows = (0, _mergeMetricsByKey.mergeMetricsByKey)([currentAssets, baselineAssets]).map(row => (0, _merge2.default)({}, row, {
    changed: (0, _getMetricChanged.getMetricChanged)(row.runs),
    added: (0, _utils.getMetricAdded)(row.runs),
    deleted: (0, _utils.getMetricDeleted)(row.runs)
  }));
  const value = (0, _utils.calculateCacheInvalidation)(rows);
  return {
    metrics: {
      cacheInvalidation: {
        value
      }
    }
  };
};

exports.extractAssetsCacheInvalidation = extractAssetsCacheInvalidation;
