// @bundle-stats/utils/lib/webpack/extract/assets-size-total-insight.js
"use strict";

require("core-js/modules/es.array.iterator");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.extractAssetsSizeTotalInsight = void 0;

var _template2 = _interopRequireDefault(require("lodash/template"));

var _get2 = _interopRequireDefault(require("lodash/get"));

var _delta = require("../../config/delta");

var _insights = require("../../config/insights");

var _metrics = require("../../utils/metrics");

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const INCREASED = (0, _template2.default)('Bundle size increased to <%= displayValue %> (<%= displayDeltaPercentage %>).');
const DECREASED = (0, _template2.default)('Bundle size decreased to <%= displayValue %> (<%= displayDeltaPercentage %>).');
const NO_CHANGE = (0, _template2.default)('Bundle size did not change.');
const TEMPLATES = new Map([[_delta.DELTA_TYPE_HIGH_NEGATIVE, INCREASED], [_delta.DELTA_TYPE_NEGATIVE, INCREASED], [_delta.DELTA_TYPE_LOW_NEGATIVE, INCREASED], [_delta.DELTA_TYPE_NO_CHANGE, NO_CHANGE], [_delta.DELTA_TYPE_LOW_POSITIVE, DECREASED], [_delta.DELTA_TYPE_POSITIVE, DECREASED], [_delta.DELTA_TYPE_HIGH_POSITIVE, DECREASED]]);
const METRIC_NAME = 'totalSizeByTypeALL';

const extractAssetsSizeTotalInsight = (webpackStats, currentExtractedData, baselineBundleStats) => {
  const currentValue = (0, _get2.default)(currentExtractedData, ['metrics', METRIC_NAME, 'value'], 0);
  const baselineValue = (0, _get2.default)(baselineBundleStats, ['metrics', 'webpack', METRIC_NAME, 'value'], 0);
  const metric = (0, _utils.getMetricType)(METRIC_NAME);
  const info = (0, _metrics.getMetricRunInfo)(metric, currentValue, baselineValue);
  const {
    deltaType,
    displayDeltaPercentage,
    displayValue
  } = info;
  const messageTemplate = TEMPLATES.get(deltaType);
  return {
    insights: {
      assetsSizeTotal: {
        type: _insights.INSIGHT_INFO,
        data: {
          text: messageTemplate({
            displayValue,
            displayDeltaPercentage
          }),
          md: messageTemplate({
            displayValue: `*${displayValue}*`,
            displayDeltaPercentage: `*${displayDeltaPercentage}*`
          }),
          info
        }
      }
    }
  };
};

exports.extractAssetsSizeTotalInsight = extractAssetsSizeTotalInsight;
