// @bundle-stats/utils/lib/browsertime/metrics.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.metrics = void 0;

var _metrics = require("../config/metrics");

const metrics = {
  firstPaint: {
    label: 'First Paint',
    type: _metrics.METRIC_TYPE_DURATION
  },
  fullyLoaded: {
    label: 'Fully Loaded',
    type: _metrics.METRIC_TYPE_DURATION
  },
  backEndTime: {
    label: 'BackEnd Time',
    type: _metrics.METRIC_TYPE_DURATION
  },
  domContentLoadedTime: {
    label: 'DOM Content Loaded Time',
    type: _metrics.METRIC_TYPE_DURATION
  },
  domInteractiveTime: {
    label: 'DOM Interactive Time',
    type: _metrics.METRIC_TYPE_DURATION
  },
  domainLookupTime: {
    label: 'Domain Lookup Time',
    type: _metrics.METRIC_TYPE_DURATION
  },
  frontEndTime: {
    label: 'FrontEnd Time',
    type: _metrics.METRIC_TYPE_DURATION
  },
  pageDownloadTime: {
    label: 'Page Download Time',
    type: _metrics.METRIC_TYPE_DURATION
  },
  pageLoadTime: {
    label: 'Page Load Time',
    type: _metrics.METRIC_TYPE_DURATION
  },
  redirectionTime: {
    label: 'Redirection Time',
    type: _metrics.METRIC_TYPE_DURATION
  },
  serverConnectionTime: {
    label: 'Server Connection Time',
    type: _metrics.METRIC_TYPE_DURATION
  },
  serverResponseTime: {
    label: 'Server Response Time',
    type: _metrics.METRIC_TYPE_DURATION
  },
  firstContentfulPaint: {
    label: 'First Contentful Paint',
    type: _metrics.METRIC_TYPE_DURATION
  },
  rumSpeedIndex: {
    label: 'RUM Speed Index',
    type: _metrics.METRIC_TYPE_DURATION
  },
  firstVisualChange: {
    label: 'First Visual Change',
    type: _metrics.METRIC_TYPE_DURATION
  },
  lastVisualChange: {
    label: 'Last Visual Change',
    type: _metrics.METRIC_TYPE_DURATION
  },
  perceptualSpeedIndex: {
    label: 'Perceptual SpeedIndex',
    type: _metrics.METRIC_TYPE_DURATION
  },
  speedIndex: {
    label: 'Speed Index',
    type: _metrics.METRIC_TYPE_DURATION
  },
  visualComplete85: {
    label: 'Visual Complete 85',
    type: _metrics.METRIC_TYPE_DURATION
  },
  visualComplete95: {
    label: 'Visual Complete 95',
    type: _metrics.METRIC_TYPE_DURATION
  },
  visualComplete99: {
    label: 'Visual Complete 99',
    type: _metrics.METRIC_TYPE_DURATION
  }
};
exports.metrics = metrics;
