// @bundle-stats/plugin-webpack-filter/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const PATH_IGNORE_PATTERN = '.map$';
exports.default = (source, options = {}) => {
    const pathIgnorePattern = new RegExp(options.pathIgnorePattern || PATH_IGNORE_PATTERN);
    const { builtAt, hash } = source;
    const assets = source.assets?.reduce((agg, asset) => {
        if (!asset.name || pathIgnorePattern.test(asset.name)) {
            return agg;
        }
        agg.push({
            name: asset.name,
            size: asset.size,
        });
        return agg;
    }, []) || [];
    const chunks = source.chunks?.reduce((agg, chunk) => {
        if (typeof chunk.id === 'undefined' || chunk.id === null) {
            return agg;
        }
        agg.push({
            id: chunk.id,
            entry: chunk.entry,
            initial: chunk.initial,
            files: chunk.files,
            names: chunk.names,
        });
        return agg;
    }, []) || [];
    const modules = source.modules?.reduce((agg, moduleStats) => {
        if (!moduleStats.name) {
            return agg;
        }
        const moduleChunks = moduleStats.chunks?.filter((chunkId) => chunkId !== null && typeof chunkId !== 'undefined') || [];
        const concatenatedModules = moduleStats.modules?.reduce((aggConcatenatedModules, concatenatedModule) => {
            if (!concatenatedModule.name) {
                return aggConcatenatedModules;
            }
            aggConcatenatedModules.push({
                name: concatenatedModule.name,
                size: concatenatedModule.size,
            });
            return aggConcatenatedModules;
        }, []);
        agg.push({
            name: moduleStats.name,
            size: moduleStats.size,
            chunks: moduleChunks,
            ...(concatenatedModules && { modules: concatenatedModules }),
        });
        return agg;
    }, []) || [];
    return {
        builtAt,
        hash,
        assets,
        chunks,
        modules,
    };
};
