// @bundle-stats/utils/lib/webpack/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SUMMARY_METRIC_PATHS = exports.SECTIONS = exports.SECTION_WEBPACK_PACKAGES = exports.SECTION_WEBPACK_MODULES = exports.SECTION_WEBPACK_ASSETS = exports.SECTION_WEBPACK_SIZES = exports.SECTION_WEBPACK_STATS = void 0;
const SECTION_WEBPACK_STATS = 'stats';
exports.SECTION_WEBPACK_STATS = SECTION_WEBPACK_STATS;
const SECTION_WEBPACK_SIZES = 'sizes';
exports.SECTION_WEBPACK_SIZES = SECTION_WEBPACK_SIZES;
const SECTION_WEBPACK_ASSETS = 'assets';
exports.SECTION_WEBPACK_ASSETS = SECTION_WEBPACK_ASSETS;
const SECTION_WEBPACK_MODULES = 'modules';
exports.SECTION_WEBPACK_MODULES = SECTION_WEBPACK_MODULES;
const SECTION_WEBPACK_PACKAGES = 'packages';
exports.SECTION_WEBPACK_PACKAGES = SECTION_WEBPACK_PACKAGES;
const SECTIONS = [SECTION_WEBPACK_STATS, SECTION_WEBPACK_SIZES, SECTION_WEBPACK_ASSETS, SECTION_WEBPACK_MODULES, SECTION_WEBPACK_PACKAGES];
exports.SECTIONS = SECTIONS;
const SUMMARY_METRIC_PATHS = ['totalSizeByTypeALL', 'totalInitialSizeJS', 'totalInitialSizeCSS', 'cacheInvalidation', 'moduleCount', 'chunkCount', 'assetCount', 'packageCount', 'duplicatePackagesCount'];
exports.SUMMARY_METRIC_PATHS = SUMMARY_METRIC_PATHS;
