// @bundle-stats/utils/lib-esm/config/delta.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DELTA_TYPE_HIGH_POSITIVE = exports.DELTA_TYPE_LOW_POSITIVE = exports.DELTA_TYPE_HIGH_NEGATIVE = exports.DELTA_TYPE_LOW_NEGATIVE = exports.DELTA_TYPE_LEVEL_HIGH = exports.DELTA_TYPE_LEVEL_LOW = exports.DELTA_TYPE_POSITIVE = exports.DELTA_TYPE_NO_CHANGE = exports.DELTA_TYPE_NEGATIVE = void 0;
const DELTA_TYPE_NEGATIVE = 'NEGATIVE';
exports.DELTA_TYPE_NEGATIVE = DELTA_TYPE_NEGATIVE;
const DELTA_TYPE_NO_CHANGE = 'NO_CHANGE';
exports.DELTA_TYPE_NO_CHANGE = DELTA_TYPE_NO_CHANGE;
const DELTA_TYPE_POSITIVE = 'POSITIVE';
exports.DELTA_TYPE_POSITIVE = DELTA_TYPE_POSITIVE;
const DELTA_TYPE_LEVEL_LOW = 'LOW';
exports.DELTA_TYPE_LEVEL_LOW = DELTA_TYPE_LEVEL_LOW;
const DELTA_TYPE_LEVEL_HIGH = 'HIGH';
exports.DELTA_TYPE_LEVEL_HIGH = DELTA_TYPE_LEVEL_HIGH;
const DELTA_TYPE_LOW_NEGATIVE = `${DELTA_TYPE_LEVEL_LOW}_${DELTA_TYPE_NEGATIVE}`;
exports.DELTA_TYPE_LOW_NEGATIVE = DELTA_TYPE_LOW_NEGATIVE;
const DELTA_TYPE_HIGH_NEGATIVE = `${DELTA_TYPE_LEVEL_HIGH}_${DELTA_TYPE_NEGATIVE}`;
exports.DELTA_TYPE_HIGH_NEGATIVE = DELTA_TYPE_HIGH_NEGATIVE;
const DELTA_TYPE_LOW_POSITIVE = `${DELTA_TYPE_LEVEL_LOW}_${DELTA_TYPE_POSITIVE}`;
exports.DELTA_TYPE_LOW_POSITIVE = DELTA_TYPE_LOW_POSITIVE;
const DELTA_TYPE_HIGH_POSITIVE = `${DELTA_TYPE_LEVEL_HIGH}_${DELTA_TYPE_POSITIVE}`;
exports.DELTA_TYPE_HIGH_POSITIVE = DELTA_TYPE_HIGH_POSITIVE;
