// @bundle-stats/utils/lib/lighthouse/metrics.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.metrics = void 0;

var _metrics = require("../config/metrics");

const metrics = {
  score: {
    label: 'Score',
    type: _metrics.METRIC_TYPE_SCORE
  },
  performanceScore: {
    label: 'Performance',
    type: _metrics.METRIC_TYPE_SCORE
  },
  accessibilityScore: {
    label: 'Accessibility',
    type: _metrics.METRIC_TYPE_SCORE
  },
  bestPracticesScore: {
    label: 'Best Practices',
    type: _metrics.METRIC_TYPE_SCORE
  },
  seoScore: {
    label: 'SEO',
    type: _metrics.METRIC_TYPE_SCORE
  },
  pwaScore: {
    label: 'PWA',
    type: _metrics.METRIC_TYPE_SCORE
  },
  speedIndex: {
    label: 'Perceptual Speed Index',
    type: _metrics.METRIC_TYPE_DURATION
  },
  firstMeaningfulPaint: {
    label: 'First Meaningful Paint',
    type: _metrics.METRIC_TYPE_DURATION
  },
  timeToFirstByte: {
    label: 'Time To First Byte',
    type: _metrics.METRIC_TYPE_DURATION
  },
  firstInteractive: {
    label: 'First Interactive',
    type: _metrics.METRIC_TYPE_DURATION
  },
  totalByteWeight: {
    label: 'Total Weight',
    type: _metrics.METRIC_TYPE_FILE_SIZE
  },
  domSize: {
    label: 'DOM Size',
    type: _metrics.METRIC_TYPE_NUMERIC
  }
};
exports.metrics = metrics;
