// @bundle-stats/utils/lib/webpack/extract/modules.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.extractModules = void 0;
const round_1 = __importDefault(require("lodash/round"));
const utils_1 = require("../utils");
const extractModules = (webpackStats) => {
    const modulesSource = webpackStats?.modules;
    if (!modulesSource) {
        return {
            metrics: {
                duplicateCode: { value: 0 },
                duplicateModulesCount: { value: 0 },
                modules: {},
                moduleCount: { value: 0 },
            },
        };
    }
    const allModules = modulesSource.reduce((agg, moduleEntry) => {
        if (!moduleEntry.modules) {
            agg.push(moduleEntry);
            return agg;
        }
        agg = agg.concat(moduleEntry.modules.map((concatenatedModule) => ({
            ...concatenatedModule,
            chunks: moduleEntry.chunks,
        })));
        return agg;
    }, []);
    let moduleCount = 0;
    let totalCodeSize = 0;
    let duplicateCodeSize = 0;
    let duplicateModulesCount = 0;
    const modules = allModules.reduce((agg, moduleEntry) => {
        const { name, size = 0, chunks } = moduleEntry;
        const normalizedName = (0, utils_1.getModuleName)(name);
        if (!chunks || chunks?.length === 0) {
            return agg;
        }
        const instances = chunks.length;
        const duplicateInstances = instances - 1;
        moduleCount += instances;
        duplicateModulesCount += duplicateInstances;
        duplicateCodeSize += duplicateInstances * size;
        totalCodeSize += instances * size;
        agg[normalizedName] = {
            name,
            value: size,
            chunkIds: chunks.map(utils_1.normalizeChunkId),
            duplicated: Boolean(duplicateInstances),
        };
        return agg;
    }, {});
    const duplicateCode = totalCodeSize ? (0, round_1.default)((duplicateCodeSize / totalCodeSize) * 100, 2) : 0;
    return {
        metrics: {
            modules,
            duplicateCode: {
                value: duplicateCode,
            },
            duplicateModulesCount: {
                value: duplicateModulesCount,
            },
            moduleCount: {
                value: moduleCount,
            },
        },
    };
};
exports.extractModules = extractModules;
